exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -36,43 +36,42 @@ class Exporter # rubocop:disable Metrics/ClassLength
-                       certificate_file: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
-                       ssl_verify_mode: Exporter.ssl_verify_mode,
-                       headers: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS', default: {}),
-                       compression: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_COMPRESSION', 'OTEL_EXPORTER_OTLP_COMPRESSION', default: 'gzip'),
-                       timeout: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: 10),
-                       metrics_reporter: nil)
-          raise ArgumentError, "invalid url for OTLP::Exporter #{endpoint}" unless OpenTelemetry::Common::Utilities.valid_url?(endpoint)
-          raise ArgumentError, "unsupported compression key #{compression}" unless compression.nil? || %w[gzip none].include?(compression)
-
-          @uri = if endpoint == ENV['OTEL_EXPORTER_OTLP_ENDPOINT']
-                   URI.join(endpoint, 'v1/traces')
-                 else
-                   URI(endpoint)
-                 end
-
-          @http = http_connection(@uri, ssl_verify_mode, certificate_file)
-
-          @path = @uri.path
-          @headers = case headers
-                     when String then parse_headers(headers)
-                     when Hash then headers
-                     else
-                       raise ArgumentError, ERROR_MESSAGE_INVALID_HEADERS
-                     end
-          @timeout = timeout.to_f
-          @compression = compression
-          @metrics_reporter = metrics_reporter || OpenTelemetry::SDK::Trace::Export::MetricsReporter
+        DEFAULT_ENDPOINT = 'http://localhost:4318/v1/traces'
+        DEFAULT_COMPRESSION = 'gzip'
+        DEFAULT_HEADERS = {}.freeze
+        DEFAULT_TIMEOUT = 10
+
+        def self.env_opts
+          {
+            endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: DEFAULT_ENDPOINT),
+            certificate_file: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE', 'OTEL_EXPORTER_OTLP_CERTIFICATE'),
+            compression: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_COMPRESSION', 'OTEL_EXPORTER_OTLP_COMPRESSION', default: DEFAULT_COMPRESSION),
+            headers: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_HEADERS', 'OTEL_EXPORTER_OTLP_HEADERS', default: DEFAULT_HEADERS),
+            timeout: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_TIMEOUT', 'OTEL_EXPORTER_OTLP_TIMEOUT', default: DEFAULT_TIMEOUT),
+            ssl_verify_mode: ssl_verify_mode,
+            metrics_reporter: nil
+          }
+        end
+
+        def initialize(opts = {})
```

Comment:
Speaking from personal experience with a variety of ruby codebases, option hashes can become quite unwieldily over time. I would not support the introduction of them in this codebase.

Reply:
I agree with you on that, what bothers me is the poor readability though. how about using named parameters as before, but defining the default values in a private method,
Something like this?
```ruby
        def initialize(endpoint: config[:endpoint], ...)
        ... 
        end

        private

        def config
        { endpoint: OpenTelemetry::Common::Utilities.config_opt('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', 'OTEL_EXPORTER_OTLP_ENDPOINT', default: DEFAULT_ENDPOINT), ... }
        end
```

Reply:
It was just as readable to me before the refactor as it is after.